subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/FileCollectionSnapshot.java

```diff
@@ -39,6 +41,8 @@
      */
     Iterator<TaskStateChange> iterateContentChangesSince(FileCollectionSnapshot oldSnapshot, String title, boolean includeAdded);
 
+    HashCode getHash();
```

Comment:
Would be nice to have some Javadoc, given that the other guys also have some.

Reply:
We could even mention that we short-circuit the comparison if we already have a hash code.